### PR TITLE
Distance_3: Add Point_3/Tetrahedron_3

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Add squared distance computation for point and tetrahedron

This replaces the PR #5304.  The proposed code here makes fewer tests.

## Release Management

* Affected package(s): Distance_3

### CODE REVIEW

File: Distance_3/include/CGAL/squared_distance_3_3.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+typename K::FT
+squared_distance(const typename K::Tetrahedron_3 & tet,
+                 const typename K::Point_3 & pt,
+                 const K& k)
+{
+  CGAL_assertion(orientation(tet[0],tet[1],tet[2],tet[3]) == POSITIVE);
+
+  if(orientation(tet[0],tet[1],tet[2],pt) != POSITIVE){
+    const Triangle_3<K> tr = {tet[0], tet[1], tet[2]};
+    return CGAL::squared_distance(tr, pt);
```
Comment (mglisse) [inline Distance_3/include/CGAL/squared_distance_3_3.h:41]: I don't think this is true. If I take a triangle, and 2 points tet[3] and pt close to each other, far from the triangle, and on opposite sides of the plane containing the triangle, this returns a large distance.

Comment (afabri) [reply Distance_3/include/CGAL/squared_distance_3_3.h:41]: you are right. I am ashamed.

**Comment by afabri** [comment]
@mglisse can you confirm that I can stop early on a triangle of the tet, when the point lies in the extruded triangle (and outside of the tet)?   This extruded triangle test is done in the triangle/point distance function.

**Comment by mglisse** [comment]
Yes, that makes more sense.
squared_distance_to_triangle could use a similar approach to compute at most 2 distances to segments, and even then you may end up computing the distance to the same segment twice (for its 2 adjacent triangles), but optimizing will be up to someone else.

File: Distance_3/include/CGAL/squared_distance_3_3.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    if(! dmin_initialized){
+      dmin = d;
+      dmin_initialized = true;
+    }else{
+      dmin = std::min(d,dmin);
+    }
+  }
+
+  if(on_bounded_side){
+    return K::FT(0);
```
Comment (maxGimeno) [inline Distance_3/include/CGAL/squared_distance_3_3.h:98]: missing typename.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-74.shtml

**Pull request merged**